### ISSUE TITLE
Version 0.10.1 gives wrap_with_ssl: undefined method empty? error

### ISSUE THREAD

**Issue by maryheintz** [ruby-ldap/ruby-net-ldap#175]
I'm having an issue with the following script.

```
#!/Users/mary/Software/rubies/2.1.4/bin/ruby
 
require 'rubygems'
gem 'net-ldap', '0.10.1'
require 'net/ldap'
 
ldap = Net::LDAP.new :host => "server.example.com",
  :port => 636,
  :encryption => { :method => :simple_tls },
  :auth => {
    :method=>:simple,
    :username=>"uid=user1,ou=people,dc=server,dc=example,dc=com",
    :password=>"password"
  }
 
if ldap.bind
 print "OK\n"
else
 print "Not OK\n"
end
```

```
yo:test $ ./check_ldap 
/Users/mary/Software/rubies/2.1.4/lib/ruby/gems/2.1.0/gems/net-ldap-0.10.1/lib/net/ldap/connection.rb:51:in `wrap_with_ssl': undefined method `empty?' for nil:NilClass (NoMethodError)
	from /Users/mary/Software/rubies/2.1.4/lib/ruby/gems/2.1.0/gems/net-ldap-0.10.1/lib/net/ldap/connection.rb:94:in `setup_encryption'
	from /Users/mary/Software/rubies/2.1.4/lib/ruby/gems/2.1.0/gems/net-ldap-0.10.1/lib/net/ldap/connection.rb:25:in `initialize'
	from /Users/mary/Software/rubies/2.1.4/lib/ruby/gems/2.1.0/gems/net-ldap-0.10.1/lib/net/ldap.rb:847:in `new'
	from /Users/maryh/Software/rubies/2.1.4/lib/ruby/gems/2.1.0/gems/net-ldap-0.10.1/lib/net/ldap.rb:847:in `block in bind'
	from /Users/maryh/Software/rubies/2.1.4/lib/ruby/gems/2.1.0/gems/net-ldap-0.10.1/lib/net/ldap/instrumentation.rb:19:in `instrument'
	from /Users/maryh/Software/rubies/2.1.4/lib/ruby/gems/2.1.0/gems/net-ldap-0.10.1/lib/net/ldap.rb:841:in `bind'
	from ./check_ldap:16:in `<main>'
```
If I change the version of net-ldap to 0.9.0, the script works fine.

Our server is running RedHat Enterprise 5.11.  Openldap version 2.3.43.

**Comment by tengchh** [comment]
I have the same issue with search function, even with 0.9.0

**Comment by jch** [comment]
@tengchh could you include a code sample that's causing the crash and also the stack trace. I want to make sure it's the same error, or if it's a separate nil issue. cc https://github.com/ruby-ldap/ruby-net-ldap/pull/168

**Comment by christianv** [comment]
@jch Thanks for this fix! Do you have an idea when the next release might happen? We would love to start using the `SimpleTLS/StartTLS encryption options` in `0.10.x`

**Comment by jch** [comment]
@christianv I have a few other pulls to review and merge, but I plan to cut a release (0.10.2) by this Friday, 1/16.

**Comment by christianv** [comment]
@jch Thanks for the update!

**Comment by tengchh** [comment]
My thanks as well.                                                                                                                                                                                                                                                                                                                                         Sent from my BlackBerry 10 smartphone.                                                                                                                                                                                                                From: Christian VueringsSent: Wednesday, January 14, 2015 18:11To: ruby-ldap/ruby-net-ldapReply To: ruby-ldap/ruby-net-ldapCc: tengchhSubject: Re: [ruby-net-ldap] Version 0.10.1 gives wrap_with_ssl: undefined method empty? error (#175)@jch Thanks for the update!

—Reply to this email directly or view it on GitHub.

**Comment by maryheintz** [comment]
Thanks!

-Mary

**Comment by christianv** [comment]
@jch do you have an update on this? Is this fixed in `0.11`?

**Comment by jch** [comment]
@christianv yes, it was fixed in #176, which is included in 0.11. https://github.com/ruby-ldap/ruby-net-ldap/blob/master/History.rdoc

**Comment by christianv** [comment]
@jch Thank you! Have a great weekend.
